### PR TITLE
Hasith/refactor/revamp landing page

### PR DESCRIPTION
**Pull request by HasithDeAlwis**
Revamp the landing-page

## Release Notes

- **New Features**
  - Updated favicon for the landing page to reflect the new logo for "cuHacking VI 2025."
  - Introduced a new `SocialsPannel` component featuring social media and documentation links with icons.

- **Improvements**
  - Simplified the `NavBar` for a cleaner layout by removing unnecessary components.
  - Added a new text style option in the `Typography` component for improved design.

- **Removals**
  - Removed the `EmailSignup` and `SponsorEmailSignup` components, eliminating email subscription functionalities to streamline the user experience.

### CODE REVIEW

File: apps/landing-page/src/pages/homePage/heroSection/EmailSignup.tsx [hunk 1]
```diff
@@ -1,131 +1,75 @@
...
+        target="_blank" 
+        rel="noopener noreferrer"
+        aria-label="LinkTree">
+          <svg xmlns="http://www.w3.org/2000/svg" shape-rendering="geometricPrecision"
+            text-rendering="geometricPrecision" image-rendering="optimizeQuality"
+            fill-rule="evenodd" clip-rule="evenodd" viewBox="0 0 417 512.238">
+            <path fill="currentColor" fill-rule="nonzero" d="M171.274 344.942h74.09v167.296h-74.09V344.942zM0 173.468h126.068l-89.622-85.44 49.591-50.985 85.439 87.829V0h74.086v124.872L331 
+            37.243l49.552 50.785-89.58 85.24H417v70.502H290.252l90.183 87.629L331 381.192 208.519 258.11 86.037 381.192l-49.591-49.591 90.218-87.631H0v-70.502z"/>
+          </svg>        
+        </a>
```
[the remark this answers is not in the document]
Comment (HasithDeAlwis) [reply apps/landing-page/src/pages/homePage/heroSection/EmailSignup.tsx:53]: Going to leave this as is, this is NOT how you should deal with SVGs IMO, and in for us right now, we just need this component made, scalability is not a concern.

File: apps/landing-page/src/pages/homePage/heroSection/SocialsPannel.tsx
Comment (Aashna-Verma) [inline apps/landing-page/src/pages/homePage/heroSection/SocialsPannel.tsx:1]: This is mighty cute! Thank you!!!!!!

**Review by JeremyFriesenGitHub — APPROVED** [review]
good stuff! coderabit was pretty useful for this and we can (if needed) apply new changes to the new refactor. Good work!

**Pull request merged**